src/TSMapEditor/Models/Map.cs

```diff
@@ -597,31 +597,34 @@ public void AddWaypoint(Waypoint waypoint)
         {
             Waypoints.Add(waypoint);
             var cell = GetTile(waypoint.Position.X, waypoint.Position.Y);
-            if (cell.Waypoint != null)
+            if (cell.Waypoints.Count > 0)
             {
-                throw new InvalidOperationException($"Cell at {cell.CoordsToPoint()} already has a waypoint, skipping adding waypoint {waypoint.Identifier}");
+                Logger.Log($"NOTE: Waypoint {waypoint.Identifier} exists in the cell {cell.CoordsToPoint()} that already has other waypoints.");
```

Comment:
Can improve this a bit (both grammar and accuracy wise).
```suggestion
                Logger.Log($"NOTE: Waypoint {waypoint.Identifier} exists in a cell {waypoint.Position} that already has one or more waypoints.");
```